### DISCUSSION TITLE
No devices and/or entities

### DISCUSSION THREAD

**Discussion by Gerlof67** [briis/unifiprotect#196] [Q&A]
Hi,

I have integrated Unifi Protect. I do see the name of the CKG2+ in the Unifi Protect-card under integrations. But I do not see any devices and or entities.
What did I do wrong?

![image](https://user-images.githubusercontent.com/78205906/108191716-c34f3f00-7113-11eb-8d97-760bde21769f.png)

**Comment by briis** [comment]
What is the FW version of your CloudKey? And what version of Unifi Protect are you running?
Have you followed the instructions in the README.md document, on how to setup a local administrator account?

**Reply by Gerlof67** [reply]
![image](https://user-images.githubusercontent.com/78205906/108196014-e16b6e00-7118-11eb-8184-0279d530271c.png)

![image](https://user-images.githubusercontent.com/78205906/108199656-a3247d80-711d-11eb-8c37-bd5c53bbed0c.png)

**Reply by Gerlof67** [reply]
I think I did it right.

**Reply by briis** [reply]
The above is the Version number of the Unfi Controller software, and not Unifi Protect. And it also does not say what FW your CloudKey.

Anyway, it seems you have the right account setup, but looking at it, it says "Never logged in". Did you use the username HAss when you setup your Unifi Protect integration?

**Reply by Gerlof67** [reply]
Yes, I did use that account.

**Reply by Gerlof67** [reply]
![image](https://user-images.githubusercontent.com/78205906/108200345-a1a78500-711e-11eb-9ba4-545f1df6184e.png)

**Reply by briis** [reply]
Could you try and click the 3 dots to the right, and click reload (or whatever it is in dutch?). And then check the log file, to see if you have anything there.

**Comment by Gerlof67** [comment]
You mean this:
![image](https://user-images.githubusercontent.com/78205906/108203758-39a76d80-7123-11eb-8be2-cae63bd5cf4a.png)

**Reply by Gerlof67** [reply]
This is in core.config_entries:

                "entry_id": "a6f6d8c526d05b8d0eeccd25f96b9d33",
                "version": 1,
                "domain": "unifiprotect",
                "title": "Cloud Key Gen2 Plus Trinks",
                "data": {
                    "id": "Cloud Key Gen2 Plus Trinks",
                    "host": "192.168.x.xx",
                    "port": 7443,
                    "username": "HAss",
                    "password": "********",
                    "snapshot_direct": false,
                    "ir_on": "auto",
                    "ir_off": "always_off",
                    "scan_interval": 2
                },
                "options": {
                    "scan_interval": 2,
                    "snapshot_direct": false
                },
                "system_options": {
                    "disable_new_entities": false
                },
                "source": "user",
                "connection_class": "local_poll",
                "unique_id": "7483C2D1ABA9"

**Reply by briis** [reply]
The issue is the port number you have used. I am pretty sure your CloudKey runs a FW greater than V1.x, and therefore it is running UnifiOS - So you need to use port 443 and NOT 7443. Try and delete the Integration, and add it again with this port.

**Reply by Gerlof67** [reply]
I can't find the current FW anyware. Only buttons to (check for) upgrades.
![image](https://user-images.githubusercontent.com/78205906/108214356-5d24e500-7130-11eb-9d04-7d675aa497af.png)


Using port 443 returns an error.
Or should I also remove the HACS integration as well?

Also, in the browser to Unifi Protect, port 7443 is used.

**Reply by briis** [reply]
Hmmm, I could recommend that you update your CloudKey+ to this version: https://community.ui.com/releases/UniFi-Cloud-Key-Firmware-2-0-27/8e72352a-dccb-437f-a5a7-34ccabcf7abf 
This will give you:
- CloudKey FW: 2.0.27
- Controller Version: 6.0.45
- Unifi Protect 1.17.2

If you do this, you might need to check that the user you created (HAss) still has admin rights, and then when you set it up, you should use port 443.

**Reply by Gerlof67** [reply]
Hmmmm. I don't like setting back firmware.
Could I get the motion-detection directly? How did you get it from the Ubiquiti camera?
That is the only thing I am missing, right now. I did get the videostream already.

**Reply by briis** [reply]
I am not convinced that the above would set you back, unless you have installed Beta Software. The link I am pointing to is the latest released version for the CloudKey. If you have already installed 1.17.3 of Unifi Protect, this will not be overwritten.

With the info you are giving, I cannot really determine if you are on V1.x or 2.x of the CloudKey FW - and that makes a huge difference, especially to the way motion detection is working. - With V1.x we have to poll the system for events, with V2.x events are pushed to HA.

**Reply by Gerlof67** [reply]
Hi,

Finally I found how to get in the Unifi Management Portal.
As can be seen, the FW is 0.9.8.

![image](https://user-images.githubusercontent.com/78205906/108722599-a0fa5e80-7523-11eb-9694-157f149c7353.png)

I can also get into it now via PuTTY.
Can you help me how to upgrade the FW?

**Reply by Gerlof67** [reply]
And upgraded to FW 1.1.13 now.
I will try to install Protect again on HA.

**Reply by briis** [reply]
Good, because that FW was really old. I will still recommend that you upgrade to the V2.0.27, as this is the latest official version, and V2.x over V1. gives a lot of advantages - especially the pushing of events instead of pulling.
Let me know how this goes.

**Reply by Gerlof67** [reply]
Now upgraded to V2.0.27.
Apparently it does upgrades step by step........

**Reply by briis** [reply]
Good. If you have the HA Integration installed, I recommend to remove it, and then add it again. (You don't have to uninstall, just remove the config). When you add again, you now need to use port 443 and not 7443.

**Comment by Gerlof67** [comment]
Better!
![image](https://user-images.githubusercontent.com/78205906/108741757-54208300-7537-11eb-9d41-faf06d36dd4d.png)
![image](https://user-images.githubusercontent.com/78205906/108741800-613d7200-7537-11eb-8e7d-95d2e6d8c873.png)

**Reply by briis** [reply]
Great, to see.

**Reply by Gerlof67** [reply]
Yeah.
Thanks for keep pushing me to the right direction!

**Reply by briis** [reply]
You are welcome. Glad it worked out.

**Reply by Gerlof67** [reply]
By the way; (related to) the cyclist......?

**Reply by briis** [reply]
No, but I do ride racing bikes - and I have more hair :)
